File: Java/Algorithms/Searching-Algorithms/BinarySearch/BinarySearch.java
class BinarySearch { 
//Binary Search Algorithm in an Iterative approach    
    int binarySearch(int arr[], int x) 
    { 
        int l = 0, r = arr.length - 1; 
        while (l <= r) { 
            int m = l + (r - l) / 2; 
  
            
            if (arr[m] == x) 
                return m; 
  
            
            if (arr[m] < x) 
                l = m + 1; 
  
            
            else
                r = m - 1; 
        } 
  
        
        return -1; 
    } 
  
//Input must be given in Ascending order (sorted)  
    public static void main(String args[]) 
    { 
        BinarySearch ob = new BinarySearch();
        int arr[] = { 2, 3, 4, 10, 40 }; 
        int n = arr.length; 
        int x = 10; 
        int result = ob.binarySearch(arr, x); 
        if (result == -1) 
            System.out.println("Element not found"); 
        else
            System.out.println("Element at "
                               + "index " + result); 
    } 
} 

File: C#/Algorithms/Searching-Algorithms/BinarySearch/BinarySearch.cs
// Implementation of a generic linear search for C#
using System;

class LinearSearch { 
	
	// Returns the index of the searched for element if present, else returns -1  
	public static int Search<T>(T[] arr, T itemToFind)
	{ 
		for (int i = 0; i < arr.Length; i++)  {
            if (arr[i].Equals(itemToFind)) {
                return i;
			}
		}

		// If this point is reached then element has not been found 
		return -1; 
	} 

	// Driver method to test above 
	public static void Main()  { 

		Console.WriteLine("Input array to search through, separated by spaces");

		var arr = Console.ReadLine();

		Console.WriteLine("Input item to search for");

		var itemToFind = Console.ReadLine();

		var result = Search(arr.Split(' '), itemToFind);

		if (result == -1) 
			Console.WriteLine("Element not present"); 
		else
			Console.WriteLine("Element found at index "
								+ result); 
	}
	
}
